azure_img_utils/cli/cli_utils.py

```diff
@@ -0,0 +1,319 @@
+    data = ChainMap(cli_values, config_values, config_defaults)
+
+    try:
+        config_data = azure_img_utils_config(**data)
+    except TypeError as e:
+        echo_style(
+            f'Found unknow keyword in config file {config_file_path}',
+            no_color=True
+        )
+        echo_style(str(e), no_color=True)
+        sys.exit(1)
+    return config_data
+
+
+# -----------------------------------------------------------------------------
+# Printing options
+def echo_style(message, no_color, fg='yellow'):
+    """
+    Echo stylized output to terminal depending on no_color.
+    """
+    if no_color:
+        click.echo(message)
+    else:
+        click.secho(message, fg=fg)
+
+
+# -----------------------------------------------------------------------------
+# Process shared options to all commands
+def process_shared_options(context_obj, kwargs):
+    """
+    Update context with values for shared options.
+    """
+    context_obj['active_directory_graph_res_url'] = \
+        kwargs.get('active_directory_graph_res_url')
+    context_obj['active_directory_url'] = kwargs.get('active_directory_url')
+    context_obj['client_id'] = kwargs.get('client_id')
+    context_obj['client_secret'] = kwargs.get('client_secret')
+    context_obj['config_dir'] = kwargs.get('config_dir')
+    context_obj['credentials_file'] = kwargs.get('credentials_file')
+    context_obj['gallery_url'] = kwargs.get('gallery_url')
+    context_obj['log_level'] = kwargs.get('log_level')
+    context_obj['management_url'] = kwargs.get('management_url')
+    context_obj['no_color'] = kwargs.get('no_color')
+    context_obj['profile'] = kwargs.get('profile')
+    context_obj['region'] = kwargs.get('region')
+    context_obj['resource_group'] = kwargs.get('resource_group')
+    context_obj['resource_manager_url'] = kwargs.get('resource_manager_url')
+    context_obj['sas_token'] = kwargs.get('sas_token')
+    context_obj['subscription_id'] = kwargs.get('subscription_id')
+    context_obj['sql_management_url'] = kwargs.get('sql_management_url')
+    context_obj['tenant_id'] = kwargs.get('tenant_id')
+
+
+# -----------------------------------------------------------------------------
+# Check required config provided
+def check_required_config_provided(config_data):
+    check_authentication_data_provided(config_data)
+
+
+# -----------------------------------------------------------------------------
+# Check required arguments for authentication provided
+def check_authentication_data_provided(config_data):
+    if config_data.sas_token:
+        return
+    elif (
+        config_data.tenant_id and
+        config_data.client_id and
+        config_data.client_secret and
+        config_data.subscription_id and
+        config_data.resource_group
+    ):
+        return
+    elif (
+        config_data.credentials_file and
+        config_data.resource_group
+    ):
+        return
+    else:
+        msg = 'Required authentication data not provided. '
```

Comment:
If this validation was kept IMHO it's nice to use brackets instead of string concatenation.

```
msg = (
    'This is a '
    'multi-line string'
)
```

Reply:
Or with slash like:

```
msg = 'This is a ' \
      'multi-line string'
```